### PR TITLE
Adds SDLJBN_AddMappingsFromRW.

### PR DESCRIPTION
**Pull request by rainerdeyke**

### CODE REVIEW

File: SDL_joystickbuttonnames.c [hunk 1]
```diff
@@ -71,15 +71,24 @@ void SDLJBN_Quit(void)
 }
 
 int SDLJBN_AddMappingsFromFile(const char *file)
+{
+	SDL_RWops *rwops = SDL_RWFromFile(file, "r");
+	if (rwops == NULL)
+	{
+		err = "Cannot open file";
```
Comment (cxong) [inline SDL_joystickbuttonnames.c:78]: need to declare and use `err`

File: SDL_joystickbuttonnames.c [hunk 2]
```diff
@@ -71,15 +71,24 @@ void SDLJBN_Quit(void)
...
+		return -1;
+	}
+	return SDLJBN_AddMappingsFromRW(rwops);
+}
+
+int SDLJBN_AddMappingsFromRW(SDL_RWops *file, int freerw)
```
Comment (cxong) [inline SDL_joystickbuttonnames.c]: rename `file` here to `rw`

**Review by cxong — CHANGES_REQUESTED** [review]
good idea to add a RW version, but please test and try the change out first. It doesn't compile

File: SDL_joystickbuttonnames.c (hunk 1 above)
Comment (rainerdeyke) [reply SDL_joystickbuttonnames.c:78]: > need to declare and use `err`

`err` is a static variable. I use it in a way that is consistent with how it is used elsewhere in the library.

Comment (cxong) [reply SDL_joystickbuttonnames.c:78]: oh yeah my bad, didn't realise it was declared already

**Pull request merged**